Sources/TuistCache/Cache/FileClient.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+import RxSwift
+import TSCBasic
+import TuistSupport
+
+enum FileClientError: LocalizedError, FatalError {
+    case urlSessionError(Error, AbsolutePath?)
+    case serverSideError(HTTPURLResponse, AbsolutePath?)
+    case invalidResponse(URLRequest, AbsolutePath?)
+    case noLocalURL(HTTPURLResponse)
+
+    // MARK: - FatalError
+
+    public var description: String {
+        var output: String
+
+        switch self {
+        case let .urlSessionError(error, path):
+            output = "Received a session error"
+            if let path = path {
+                output.append(" while uploading file at path \(path.pathString)")
+            }
+            if let error = error as? LocalizedError {
+                output.append("\nError: \(error.localizedDescription)")
+            }
+        case let .invalidResponse(urlRequest, path):
+            output = "Received unexpected response from the network with url request \(urlRequest)"
```

Comment:
Sorry for being nitpicking about this, but what we print to the users in their terminal is like the views in the app. What I mean by that is that we have to make sure that the contains enough information for the user to know what's going on, but without including too much verbose information in it. In this case printing `\(urlRequest)` might result in a lot of verbose information that is not relevant to the user.

I'd recommend printing something along the lines of:

```
Received unexpected response from the network after sending a POST request to https://....`
```

Reply:
Addressed.